backend/pkg/helm/repository.go

```diff
@@ -0,0 +1,265 @@
+package helm
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	zlog "github.com/rs/zerolog/log"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	defaultNewConfigFileMode   os.FileMode = os.FileMode(0644)
+	defaultNewConfigFolderMode os.FileMode = os.FileMode(0770)
+)
+
+// add repository
+
+type AddUpdateRepoRequest struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+	// TODO: Figure out how to support auth
+	// like username, password, certfile etc
+	// https://github.com/helm/helm/blob/39ca699ca790e02ba36753dec6ba4177cc68d417/cmd/helm/repo_add.go#L169
+}
+
+// Creates a filename if it's not there, including any missing directories.
+func createFileIfNotThere(fileName string) error {
+	_, err := os.Stat(fileName)
+	if os.IsNotExist(err) {
+		// create changes
+		_, err = createFullPath(fileName)
+		return err
+	}
+
+	return nil
+}
+
+// Adds a repository with name, url to the helm config. Returns error if there is one.
+func AddRepositoryToHelm(name string, url string) error {
+	err := createFileIfNotThere(settings.RepositoryConfig)
+	if err != nil {
+		zlog.Error().Err(err).Str("action", "add_repo").Msg("failed to create empty RepositoryConfig file")
+		return err
+	}
+
+	// TODO: Lock repo file
+
```

Comment:
As well as file locking, missing from here is checking if the repo exists already. As is done here: 
 https://github.com/helm/helm/blob/39ca699ca790e02ba36753dec6ba4177cc68d417/cmd/helm/repo_add.go#L102

Reply:
Since this is fixed now, resolving the dicussion.